src/main/java/pages/TodaySDealsPage.java

```diff
@@ -0,0 +1,74 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.List;
+
+public class TodaySDealsPage extends BasePage {
+
+    public TodaySDealsPage(WebDriver driver) {
+        super(driver);
+    }
+
+    @FindBy(linkText = "Select All")
+    protected WebElement selectAllButton;
+    @FindBy(css = "input[type='checkbox']")
+    protected List<WebElement> listOfFilterCheckboxes;
+    @FindBy(linkText = "Clear")
+    protected WebElement clearButton;
+
+    public String getTextFromSelectAllButton() {
+        return selectAllButton.getText();
+    }
+
+    protected String inputSelector = "//span[text()='%s']/parent::label/input";
+
+    public void selectTwoFilterItems(String filterItemOne, String filterItemTwo) {
+        driver.findElement(By.xpath(String.format(inputSelector, filterItemOne))).click();
+        driver.findElement(By.xpath(String.format(inputSelector, filterItemTwo))).click();
```

Comment:
Overall this is OK, but just a note, that filterIItem texts can be the same and your test will fail then.
But overall, this is OK.